### ISSUE TITLE
[meta] Upgrade integrations to ECS 8.17

### ISSUE THREAD

**Issue by taylor-swanson** [elastic/integrations#11952]
## Guide

It is recommended to split the changes into smaller PRs to limit the number of changed files. Generally speaking, 10 integrations per PR is a good number to target.

### Automated method

Use the `ecs-update` tool which can be [found here](https://github.com/andrewkroh/go-examples/tree/main/ecs-update).

```
cd packages/
ecs-update -pr 999999 -ecs-git-ref v8.17.0 -ecs-version 8.17.0 -owner elastic/owner-name-here package_1 [package_2 ...]
```

Once PR is filed, the changelogs will need to be updated with the correct PR number.

### Manual method

Update ECS references in integrations to version 8.17.

- Update reference in `_dev/build/build.yml` to `v8.17.0`
- Update package changelog/manifest and regenerate README.

```diff
dependencies:
  ecs:
-    reference: "git@v8.11.0"
+    reference: "git@v8.17.0"
```

## ECS 8.17 Changes

https://github.com/elastic/ecs/releases/tag/v8.17.0

#### Bugfixes

* Fix link rendering issues and usage of http in links. #2423

#### Improvements

* Increase ignore_above value for url.query. #2424
* Set synthetic_source_keep = none on fields that represent sets. #2422

### @elastic/cloud-security-posture:
- [ ] cloud_security_posture

### @elastic/ecosystem:
- [ ] elastic_package_registry

### @elastic/elastic-agent:
- [ ] elastic_agent

### @elastic/elastic-agent-data-plane: 
- [ ] filestream
- [ ] journald
- [ ] linux
- [ ] log
- [ ] windows

### @elastic/fleet:
- [ ] fleet_server

### @elastic/obs-cloudnative-monitoring:
- [ ] containerd
- [ ] docker
- [ ] istio
- [ ] kubernetes
- [ ] kubernetes_otel
- [ ] nginx_ingress_controller

### @elastic/obs-ds-hosted-services: 
- [ ] aws
- [ ] aws_logs
- [ ] awsfirehose
- [ ] azure
- [ ] azure_metrics
- [ ] gcp
- [ ] gcp_metrics

### @elastic/obs-ds-intake-services: 
- [ ] apm
- [ ] profiler_collector
- [ ] profiler_symbolizer

### @elastic/obs-infraobs-integrations:
- [ ] activemq
- [x] airflow https://github.com/elastic/integrations/pull/15287
- [ ] apache
- [ ] apache_spark
- [ ] apache_tomcat
- [ ] awsfargate
- [ ] azure_app_service
- [ ] azure_application_insights
- [ ] azure_billing
- [ ] azure_functions
- [ ] azure_logs
- [ ] azure_openai
- [ ] cassandra
- [ ] ceph
- [ ] cisco_meraki_metrics
- [ ] citrix_adc
- [ ] cockroachdb
- [ ] coredns
- [ ] couchbase
- [ ] couchdb
- [ ] etcd
- [ ] gcp_vertexai
- [ ] golang
- [ ] hadoop
- [ ] haproxy
- [ ] ibmmq
- [ ] iis
- [ ] influxdb
- [ ] jolokia
- [ ] kafka
- [ ] kafka_log
- [ ] memcached
- [ ] microsoft_sqlserver
- [ ] mongodb
- [ ] mongodb_atlas
- [ ] mysql
- [ ] nagios_xi
- [ ] nats
- [ ] nginx
- [ ] nginx_ingress_controller_otel
- [ ] oracle
- [ ] oracle_weblogic
- [ ] panw_metrics
- [ ] php_fpm
- [ ] postgresql
- [ ] prometheus
- [ ] prometheus_input
- [ ] rabbitmq
- [ ] redis
- [ ] redisenterprise
- [ ] salesforce
- [ ] spring_boot
- [ ] sql
- [ ] stan
- [x] statsd_input https://github.com/elastic/integrations/pull/15125
- [ ] system
- [ ] tomcat
- [ ] traefik
- [ ] vsphere
- [ ] websphere_application_server
- [ ] zookeeper

### @elastic/sec-deployment-and-devices:

- #12569
  - [x] arista_ngfw
  - [x] cef
  - [x] checkpoint
  - [x] cisco_aironet
  - [x] cisco_asa
  - [x] cisco_ftd
  - [x] cisco_ios
  - [x] cisco_ise
  - [x] cisco_nexus
  - [x] cisco_secure_email_gateway
  - [x] citrix_waf

- #12571  
  - [x] fortinet_forticlient
  - [x] fortinet_fortiedr
  - [x] fortinet_fortigate
  - [x] fortinet_fortimail
  - [x] fortinet_fortimanager
  - [x] fortinet_fortiproxy
  - [x] goflow2
  - [x] hashicorp_vault
  - [x] imperva
  - [x] iptables

- #12572 
  - [x] juniper_srx
  - [x] modsecurity
  - [x] netflow
  - [x] osquery
  - [x] panw
  - [x] pfsense
  - [x] proxysg

- #12574 
  - [x] qnap_nas
  - [x] snort
  - [x] sonicwall_firewall
  - [x] sophos
  - [x] squid
  - [x] stormshield
  - [x] suricata
  - [x] tcp
  - [x] udp
  - [x] watchguard_firebox
  - [x] zeek

### @elastic/sec-linux-platform:
- [ ] auditd
- [ ] auditd_manager
- [ ] cloud_defend
- [ ] fim
- [ ] network_traffic
- [ ] sysmon_linux
- [ ] system_audit

### @elastic/sec-windows-platform: https://github.com/elastic/integrations/pull/12636
- [x] hid_bravura_monitor
- [x] microsoft_dhcp
- [x] microsoft_dnsserver
- [x] microsoft_exchange_server
- [x] mysql_enterprise
- [x] windows_etw
- [x] winlog

### @elastic/security-asset-management:
- [ ] osquery_manager

### @elastic/security-service-integrations:
- #14158
  - [x] claroty_ctd
  - [x] entityanalytics_ad
  - [x] sysdig
  - [x] tychon
 
- [ ] 1password
- [ ] abnormal_security
- [ ] akamai
- [ ] amazon_security_lake
- [ ] atlassian_bitbucket
- [ ] atlassian_confluence
- [ ] atlassian_jira
- [ ] auth0
- [ ] authentik
- [ ] aws_bedrock
- [ ] azure_blob_storage
- [ ] azure_frontdoor
- [ ] azure_network_watcher_nsg
- [ ] azure_network_watcher_vnet
- [ ] barracuda
- [ ] barracuda_cloudgen_firewall
- [ ] bbot
- [ ] bitdefender
- [ ] bitwarden
- [ ] blacklens
- [ ] box_events
- [ ] canva
- [ ] carbon_black_cloud
- [ ] carbonblack_edr
- [ ] cel
- [ ] checkpoint_harmony_endpoint
- [ ] cisa_kevs
- [ ] cisco_duo
- [ ] cisco_meraki
- [ ] cisco_secure_endpoint
- [ ] cisco_umbrella
- [ ] cloudflare
- [ ] cloudflare_logpush
- [ ] corelight
- [ ] cribl
- [ ] crowdstrike
- [ ] cyberark_pta
- [ ] cyberarkpas
- [ ] cybereason
- [ ] cylance
- [ ] darktrace
- [ ] digital_guardian
- [ ] entityanalytics_entra_id
- [ ] entityanalytics_okta
- [ ] eset_protect
- [ ] f5
- [ ] f5_bigip
- [ ] falco
- [ ] fireeye
- [ ] first_epss
- [ ] forcepoint_web
- [ ] forgerock
- [ ] gcp_pubsub
- [ ] gigamon
- [ ] github
- [ ] gitlab
- [ ] google_cloud_storage
- [ ] google_scc
- [ ] google_workspace
- [ ] http_endpoint
- [ ] httpjson
- [ ] imperva_cloud_waf
- [ ] infoblox_bloxone_ddi
- [ ] infoblox_nios
- [ ] jamf_compliance_reporter
- [ ] jamf_pro
- [ ] jamf_protect
- [ ] jumpcloud
- [ ] keycloak
- [ ] lastpass
- [ ] lumos
- [ ] lyve_cloud
- [ ] m365_defender
- [ ] mattermost
- [ ] menlo
- [ ] microsoft_defender_cloud
- [ ] microsoft_defender_endpoint
- [ ] microsoft_exchange_online_message_trace
- [ ] microsoft_sentinel
- [ ] mimecast
- [ ] netskope
- [ ] o365
- [ ] okta
- [ ] opencanary
- [ ] panw_cortex_xdr
- [ ] ping_one
- [ ] pps
- [ ] prisma_access
- [ ] prisma_cloud
- [ ] proofpoint_on_demand
- [ ] proofpoint_tap
- [ ] pulse_connect_secure
- [ ] qualys_vmdr
- [ ] rapid7_insightvm
- [ ] santa
- [ ] sentinel_one
- [ ] sentinel_one_cloud_funnel
- [ ] servicenow
- [ ] slack
- [ ] snyk
- [ ] sophos_central
- [ ] spycloud
- [ ] sublime_security
- [ ] symantec_edr_cloud
- [ ] symantec_endpoint
- [ ] symantec_endpoint_security
- [ ] tanium
- [ ] teleport
- [ ] tenable_io
- [ ] tenable_sc
- [ ] threat_map
- [ ] thycotic_ss
- [ ] ti_abusech
- [ ] ti_anomali
- [ ] ti_cif3
- [ ] ti_crowdstrike
- [ ] ti_custom
- [ ] ti_cybersixgill
- [ ] ti_eclecticiq
- [ ] ti_eset
- [ ] ti_maltiverse
- [ ] ti_mandiant_advantage
- [ ] ti_misp
- [ ] ti_opencti
- [ ] ti_otx
- [ ] ti_rapid7_threat_command
- [ ] ti_recordedfuture
- [ ] ti_threatconnect
- [ ] ti_threatq
- [ ] ti_util
- [ ] tines
- [ ] trellix_edr_cloud
- [ ] trellix_epo_cloud
- [ ] trend_micro_vision_one
- [ ] trendmicro
- [ ] vectra_detect
- [ ] websocket
- [ ] wiz
- [ ] zerofox
- [ ] zeronetworks
- [ ] zoom
- [ ] zscaler_zia
- [ ] zscaler_zpa

### @elastic/stack-monitoring:
- [ ] beat
- [ ] elasticsearch
- [ ] enterprisesearch
- [ ] kibana
- [ ] logstash
- [ ] platform_observability

**Comment by consulthys** [comment]
> ### [@elastic/stack-monitoring](https://github.com/orgs/elastic/teams/stack-monitoring):
> * [ ]  beat[ ]  elasticsearch[ ]  enterprisesearch[ ]  kibana[ ]  logstash[ ]  platform_observability

Hey @taylor-swanson thanks for initiating this. What is the deadline for this change?
Also, can you share whether this should be done before or after https://github.com/elastic/ingest-dev/issues/4484 ?

**Comment by taylor-swanson** [comment]
Hey @consulthys!

While it is nice to have ECS updated, it isn't critical it needs to be done immediately, unless an integration needs to take advantage of a change in the new version. We haven't issued a new ECS release in a while (8.11.0 was the previous version), but at least back then, we'd work on getting the integrations updated within a short period of time (perhaps a week or so?) after the release.

The `ecs-update` tool (see the issue description above) definitely helps make the process of updating integrations less painful.

Regarding how it relates to the issue you linked, that's actually a good question. As far as I know, I don't think 9.0 will have any issues using a 8.16 (or any 8.x version) ECS. There isn't a strong version enforcement like there is for `kibana.version` in a  package manifest. I'll raise that question with the ECS team, though. If we do need a 9.0 version, we'll for sure get another issue out like this one.

**Comment by taylor-swanson** [comment]
After discussing with the ECS working group, we've decided to hold off on upgrading to 8.16.0 due to link rendering issues. We will be fixing these issues for 8.17.0 and then work can resume on upgrading integrations. If the integrations you are working on don't have any issues with links and you would still like to upgrade to 8.16.0, that is okay.

I don't have a date for the ECS release, but I'd imagine it's relatively soon given that the 8.17 stack is already out.
